src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,74 @@
+package gift.controller;
+
+import gift.dto.RequestDto;
+import gift.dto.ResponseDto;
+import gift.entity.Product;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+    private final Map<Long, Product> products = new HashMap<>();
+
+    // 1. 상품 추가
+    @PostMapping
+    public ResponseEntity<ResponseDto> createProduct(@RequestBody RequestDto dto) {
+        Long productId = products.isEmpty() ? 1 : Collections.max(products.keySet()) + 1;
+
+        Product product = new Product(productId, dto.getName());
+
+        products.put(productId, product);
+
+        return new ResponseEntity<>(new ResponseDto(product), HttpStatus.CREATED);
+    }
+
+    // 2. 상품 조회
+    @GetMapping("/{id}")
+    public ResponseEntity getProduct(@PathVariable Long id) {
+        Product product = products.get(id);
+
+        if (product == null) {
+            return new ResponseEntity(HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity(new ResponseDto(product), HttpStatus.OK);
+    }
```

Comment:
```suggestion
    @GetMapping("/{id}")
    public ResponseEntity<ResponseDto> getProduct(@PathVariable Long id) {
        Product product = products.get(id);

        if (product == null) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }

        return new ResponseEntity<>(new ResponseDto(product), HttpStatus.OK);
    }
```
위에서 하신 것과 마찬가지로 제네릭 타입을 명시하면 좋을 것 같습니다~
이걸 하면 어떤 부분이 좋은 것일까요?